From 09e2e5c0c8e3ff08d562ce8704b00bc285a487e9 Mon Sep 17 00:00:00 2001
From: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Date: Thu, 7 Jun 2018 15:59:52 +0300
Subject: [PATCH 3946/5242] MLK-16784 sound: asoc: mark micfil stat register
 volatile

commit  89e3d57be4017c486b32689fd2e8b89bc3c759cc from
https://source.codeaurora.org/external/imx/linux-imx.git

Mark micfil stat register as volatile since status
flags should not be read from cache.

Signed-off-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_micfil.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/fsl/fsl_micfil.c b/sound/soc/fsl/fsl_micfil.c
index 47953f0..5eb3a91 100644
--- a/sound/soc/fsl/fsl_micfil.c
+++ b/sound/soc/fsl/fsl_micfil.c
@@ -617,6 +617,7 @@ static bool fsl_micfil_volatile_reg(struct device *dev, unsigned int reg)
 {
 	switch (reg) {
 	case REG_MICFIL_CTRL1:
+	case REG_MICFIL_STAT:
 	case REG_MICFIL_DATACH0:
 	case REG_MICFIL_DATACH1:
 	case REG_MICFIL_DATACH2:
-- 
1.7.9.5

